src/ray/raylet/main.cc

```diff
@@ -193,13 +195,51 @@ int main(int argc, char *argv[]) {
   RAY_CHECK_OK(gcs_client->Connect(main_service, cluster_id));
   std::unique_ptr<ray::raylet::Raylet> raylet;
 
+  // Enable subreaper. This is called in `AsyncGetInternalConfig` below, but MSVC does
+  // not allow a macro invocation (#ifdef) in another macro invocation (RAY_CHECK_OK),
+  // so we have to put it here.
+  auto enable_subreaper = [&]() {
+#ifdef __linux__
```

Comment:
This probably is not an issue. Maybe we can put the __linux__ into the SetThisProcessAsSubreaper. 

For the platform specific code, the lower level we put it, the better.